pkg/validation/validate_cluster.go

```diff
@@ -96,6 +109,41 @@ func getRoleNode(node *v1.Node) string {
 	return role
 }
 
+func collectComponentStatus(client k8s_clientset.Interface) (bool, error) {
+	componentList, err := client.CoreV1().ComponentStatuses().List(v1.ListOptions{})
+	if err != nil {
+		fmt.Println("first error")
+		return false, err
+	}
+
+	fmt.Println("looop")
+	for _, component := range componentList.Items {
+		for _, condition := range component.Conditions {
+			if condition.Status != "True" {
```

Comment:
You should probably use the strongly-typed `api.ConditionTrue` instead of "True" here.